src/entitysdk/downloaders/morphology.py

```diff
@@ -0,0 +1,55 @@
+"""Download functions for Morphology entities."""
+
+import logging
+import pathlib
+
+logger = logging.getLogger(__name__)
+
+
+def download_morphology(client, morphology, morph_dir="./morphology", file_type="asc"):
+    """Download morphology file.
+
+    Args:
+        client (Client): EntitySDK client
+        morphology (ReconstructionMorphology): Morphology entitysdk object
+        morph_dir (str or Pathlib.Path): directory to save the morphology file
+        file_type (str or None): type of the morphology file (asc, swc or h5).
+            Will take the first one if None.
+    """
+    if not morphology.assets:
+        raise ValueError(f"No assets found in the morphology {morphology.name}.")
+    morph_dir = pathlib.Path(morph_dir)
+    morph_dir.mkdir(parents=True, exist_ok=True)
+
+    # try to fetch morphology with the specified file type
+    asset = client.download_assets(
+        morphology,
+        selection={"content_type": f"application/{file_type}"},
+        output_path=morph_dir,
+    ).one_or_none()
+    # fallback #1: we expect at least a asc or swc file
+    if asset is None:
+        asset = client.download_assets(
+            morphology,
+            selection={"content_type": "application/asc"},
+            output_path=morph_dir,
+        ).one_or_none()
+        if asset is None:
+            asset = client.download_assets(
+                morphology,
+                selection={"content_type": "application/swc"},
+                output_path=morph_dir,
+            ).one_or_none()
+    # fallback #2: we take the first asset
+    if asset is None:
+        logger.warning(
+            "No %s file found in the morphology %s, will select the first one.",
```

Comment:
This should not happen because what was requested with the signature of the function is not the result.

Is there a point downloading a random asset other than the one needed by the code?

imo if the requested content type is not there, it should throw